### ISSUE TITLE
Supporting Org Mode file Syntax highlighting

### ISSUE THREAD

**Issue by mmahmoudian** [rstudio/rstudio#10309]
**Disclaimer:** I searched among the issues and didn't find any thing that matches ["orgmode"](https://github.com/rstudio/rstudio/issues?q=is%3Aissue+%22orgmode%22+) or ["org mode"](https://github.com/rstudio/rstudio/issues?q=is%3Aissue+%22org+mode%22) and therefore I'm filling this feature request.

![image](https://user-images.githubusercontent.com/390889/148231174-280e6adf-be07-40a8-afc7-0ef67e4a4733.png)

## Feature request

There are two major markups that are frequently used by programmers and researchers: Markdown (and it's flavors) and OrgMode. The current version of Rstudio (`RStudio 2021.09.1+372 "Ghost Orchid"`) supports Markdown among their files and the OrgMode is very much missed imho and it would be nice to have it supported as well:

![image](https://user-images.githubusercontent.com/390889/148230408-9c600f33-0ea7-4d4b-9bd8-ba02062a8822.png)

There are two layers in which this can be implemented:
1. Only syntax highlighting
2. Full Org Mode (https://orgmode.org/)

Of course it would be totally nice to have the full Org Mode as Rstudio already supports Emacs keybindings, but having the syntax highlighting would be already a very good start.

**Comment by kevinushey** [comment]
Can you share an example of what syntax elements you'd like to see highlighted? A screenshot might be useful as well.

**Comment by mmahmoudian** [comment]
I copied the exact sample text from the orgmode.org and pasted it in Rstudio and in Emacs , resized the windows to show the same line in the same height:

![image](https://user-images.githubusercontent.com/390889/148548600-a2f47188-cb55-46a1-969d-f4a346a8d651.png)

Note that I have put the Rstudio on Markdown mode as some parts of the syntax are the same (the best possible compatibility I could think of).

**Comment by kevinushey** [comment]
Thanks for the detailed explanation! I think some of these could be welcome enhancements to our default Markdown highlight rules, as well.

**Comment by mmahmoudian** [comment]
I agree that full org-mode support would be a bit too much, but the syntax highlighting and folding would be awesome. Additionally the source blocks would be probably easy to add as the Rmarkdown has the chunks already, but that is just some extra touch imho.
